core/orchestrator.go

```diff
@@ -842,12 +852,19 @@ func (rtm *RemoteTranscoderManager) Manage(stream net.Transcoder_RegisterTransco
 		totalLoad, totalCapacity, liveTranscodersNum = rtm.totalLoadAndCapacity()
 	}
 	rtm.RTmutex.Unlock()
+
+	// Remove all streamSessions assigned to this transcoder
+	for sessionId, assignedTranscoder := range rtm.streamSessions {
```

Comment:
Yup, this would certainly get more expensive with the increase in the # of streams, although even if we were to handle it in `selectTranscoder()`, we would have to loop over all the sessions, since it's a mapping of streamSession => T to remove all sessions for a particular T. Also, I suppose that would increase the response time of `selectTranscoder()`, which is more critical than the speed at which an EOF is handled. Is that right?

Reply:
My thought was that we can avoid the loop all together. Instead of removing all sessions for a T in a single loop when the T disconnects, leave the sessions intact. Then, when `selectTranscoder()` is called, the for loop will make sure that transcoders that are no longer in the live map are skipped and then a session can be removed at that point:

```golang
for checkTranscoders(rtm) {
    ...
    if _, ok := rtm.liveTranscoders[currentTranscoder.stream]; !ok {
        // Remove the stream session because the transcoder is no longer live
        if sessionExists {
            rtm.completeStreamSession(sessionId)
        }
        ...
    }
}
```